tests/test_dirs.py

```diff
@@ -342,10 +343,6 @@ def create_file_build_dir_is_returned_as_created(build_dir: BuildDir) -> Iterabl
         create_file_return_nonexisting_relative_path,
         pytest.raises(FileNotFoundError, match="does not exist inside build directory"),
     ],
-    [
-        create_file_linked_to_outside_file,
-        pytest.raises(ValueError, match="File must be created inside"),
```

Comment:
Hmm, maybe we should still check this? Don't follow broken symlinks, but do protect against potentially malicious ones?

Reply:
I am not sure what you mean, because the test is still checking the same thing, except now it does not fail.

Reply:
```
.
├── broken-link -> nonexistent-file
├── circular-link -> circular-link2
├── circular-link2 -> circular-link
└── malicious-link -> /etc/shadow
```
I think we should allow broken-link and circular-link but not malicious-link. The test was previously checking the malicious-link case, I don't think we had tests for the other two

Reply:
So, do you mean that:

1. that we should still raise error when symlink points to a path outside of build_dir
2. allow broken links, circular links, working links that point to files in build_dir (but still copy only the symlink)

Reply:
Although we won't be checking malicious links in subdirectories anyway, so it doesn't matter that much.

Basically, my point was that we still want this line
```python
file_path = file_path.resolve()
```
But that's not that important. The more important point is that we still don't have a test case for broken symlinks :smile:

Reply:
Definitely 2. (that's already solved, just needs the test), 1. matters less

Reply:
Btw, it seems that creating a broken symlink can be done like this
```python
Path("broken-link").symlink_to("doesnt-exist")
```

Reply:
Ahh, I forgot about the `exists` check. Broken symlinks won't work if we keep that check. 

But keep in mind that we only check files which were directly returned by the function passed to for_all_platforms. If the function returns a directory, we check the directory itself but not anything inside. And that's really the main thing we want to fix (for the Cachito use case).

It's probably OK to fail for broken symlinks returned directly

Reply:
there is no other way, because even if the exists check was removed, then it would fail on executing copy method. So I'll try to incorporate unit test that creates dir with broken symlinks inside.

Reply:
I believe that now it should be ok. Although it seems strange to me that we are treating files in a directory differently.